### DISCUSSION TITLE
데이터 클래스를 어떤 모듈에 넣어야 할지 질문합니다.

### DISCUSSION THREAD

**Discussion by choius323** [boostcampwm-2022/android08-KiwiTalk#38] [멘토님께 질문]
구글 지도를 사용 중에 기준에 맞춘 데이터 클래스를 만들어야 사용할 수 있는 기능이 있습니다.
그런데 app 모듈에서 사용하는 데이터 클래스를 domain 모듈에 넣고 사용하려고 했는데 새로운 클래스를 어디에 넣어야 할지 의문이 들어서 질문합니다.

```kotlin
import com.google.android.gms.maps.model.LatLng
import com.google.maps.android.clustering.ClusterItem
import com.kiwi.domain.model.Marker

data class ClusterMarker(
    private val _latLng: LatLng,
    private val _snippet: String?,
    private val _title: String?
) : ClusterItem {
    val x = _latLng.latitude
    val y = _latLng.longitude
    override fun getPosition(): LatLng = _latLng
    override fun getSnippet(): String? = _snippet
    override fun getTitle(): String? = _title

    constructor(x: Double, y: Double, snippet: String?, title: String?) :
            this(LatLng(x, y), snippet, title)

    companion object {
        fun Marker.toMyItem(): ClusterMarker = ClusterMarker(
            x = this.x,
            y = this.y,
            snippet = toString(),
            title = cid
        )
    }
}
```
[원본 링크](https://github.com/boostcampwm-2022/android08-KiwiTalk/blob/bcb2fe439b49ad4c34989f884b53d8781bd4c1c1/app/src/main/java/com/kiwi/kiwitalk/model/ClusterMarker.kt)

상속하고 있는 `ClusterItem` 인터페이스를 구현해야 쓸 수 있는 기능이 있는데 이 클래스를 domain 모듈에 넣으려면 domain에 android, google map 의존성을 추가해야 하는 상황입니다.

현재는 app-model 클래스에 들어가 있고 `data - MarkerRemote` -> `domain - Marker` -> `app - ClusterMarker` 형태로 총 3개의 클래스를 사용해서 데이터를 전달하고 있습니다.

이럴 때는 어떤 모듈에 넣고 데이터 형식 변환을 어떻게 하는 게 좋을까요?

**Comment by KimGyeongyeon** [comment]
1. app 모듈에 클래스가 추가로 있는 것은 문제가 없다. domain이 특정 라이브러리에 의존성이 있으면 안되기 때문이다.
2. 변수에 _가 있는 것이 예쁘지 않다. 대신 latLng, snippet, title
3. MarkerRemote는 intenal로 만들면 관심사 분리 강화된다.
